packages/squiggle-lang/src/rescript/Utility/E.res

```diff
@@ -289,8 +289,14 @@ module A = {
       ))
       |> Rationale.Result.return
     }
-  let rangeFloat = (~step=1, start, stop) =>
-    Belt.Array.rangeBy(start, stop, ~step) |> fmap(Belt.Int.toFloat)
+
+  let diff = (arr: array<float>): array<float> =>
+    Belt.Array.zipBy(arr, Belt.Array.sliceToEnd(arr, 1), (left, right) => right -. left)
+
+  let rec rangeFloat = (~step: float=1.0, start : float, end: float) : array<float> => 
+    start > end ?
+      []
+      : Belt.Array.concat([start], rangeFloat(~step, start +. step, end))
```

Comment:
Could the above use ``reduce`` instead of recursion? I generally avoid recursion.

Reply:
Turns out what I wanted you already implemented, so I removed this